### PR TITLE
Make -c option append test results

### LINKED ISSUE #23: Multiple test results not aggregated properly in report command
Let's say we have 1 passed test run and 1 timeout.
Running a report on a single passed run results in an expected report.
```
[ivasilev@localhost tesar]$ python tesar/__main__.py report -c http://artifacts.osci.redhat.com/testing-farm/14612a82-002d-4a8d-a5c4-613a0a75efeb
Reporting for the requested tasks:
RHEL-8.8.0-Nightly tier1/upgrade_happy_path https://api.dev.testing-farm.io/v0.1/requests/14612a82-002d-4a8d-a5c4-613a0a75efeb COMPLETE
+--------------------------------------+--------------------+--------------------------------------+--------+
| UUID                                 | Target             | Test Plan                            | Result |
+--------------------------------------+--------------------+--------------------------------------+--------+
| 14612a82-002d-4a8d-a5c4-613a0a75efeb | RHEL-8.8.0-Nightly |                                      |        |
|                                      |                    | /plans/8to9/tier1/upgrade_happy_path | PASSED |
+--------------------------------------+--------------------+--------------------------------------+--------+
```
While running a report command on 2 runs will result either in "timeout has occured" or "1 test passed" (depending on the order of arguments) with no information about the second run.
```
[ivasilev@localhost tesar]$ python tesar/__main__.py report -c http://artifacts.osci.redhat.com/testing-farm/14612a82-002d-4a8d-a5c4-613a0a75efeb -c http://artifacts.osci.redhat.com/testing-farm/909143cf-89ca-4d62-8f81-959bf8ab4d03/
Reporting for the requested tasks:
RHEL-8.6.0-Nightly ^(?!.*upgrade_plugin)(?!.*tier[2-3].*)(?!.*rhsm)(?!.*c2r)(?!.*sap)(?!.*7to8)(?!.*max_sst) https://api.dev.testing-farm.io/v0.1/requests/909143cf-89ca-4d62-8f81-959bf8ab4d03 ERROR
Request ended up in ERROR state, because Pipeline timeout expired., thus won't be reported.
See more details on the result page http://artifacts.osci.redhat.com/testing-farm/909143cf-89ca-4d62-8f81-959bf8ab4d03
Skipping to next request.
Nothing to report!
```
```
[ivasilev@localhost tesar]$ python tesar/__main__.py report -c http://artifacts.osci.redhat.com/testing-farm/909143cf-89ca-4d62-8f81-959bf8ab4d03/ -c http://artifacts.osci.redhat.com/testing-farm/14612a82-002d-4a8d-a5c4-613a0a75efeb 
Reporting for the requested tasks:
RHEL-8.8.0-Nightly tier1/upgrade_happy_path https://api.dev.testing-farm.io/v0.1/requests/14612a82-002d-4a8d-a5c4-613a0a75efeb COMPLETE
+--------------------------------------+--------------------+--------------------------------------+--------+
| UUID                                 | Target             | Test Plan                            | Result |
+--------------------------------------+--------------------+--------------------------------------+--------+
| 14612a82-002d-4a8d-a5c4-613a0a75efeb | RHEL-8.8.0-Nightly |                                      |        |
|                                      |                    | /plans/8to9/tier1/upgrade_happy_path | PASSED |
+--------------------------------------+--------------------+--------------------------------------+--------+
```

### PR DESCRIPTION
**Pull request by fernflower**
Instead of having -c accept a list of tests like '-c test1 test2 test3' let's have a more natural way of '-c test1 -c test2 -c test3'. Covered by a unit test.

Closes: #23

### CODE REVIEW

**Comment by fernflower** [comment]
Example of github action run - https://github.com/fernflower/tesar/actions/runs/5725710179/job/15514717895?pr=1 

Imho it's good enough to be reviewed @danmyway  :)

File: src/dispatch/__init__.py [hunk 1]
```diff
@@ -380,29 +296,29 @@ def get_arguments():
...
     )
 
-    args = parser.parse_args()
+    args = parser.parse_args(args=args)
     return args
 
 
-ARGS = get_arguments()
+def get_compose_mapping(args=sys.argv[1:]):
+    args = get_arguments(args=args)
```
Comment (danmyway) [inline src/dispatch/__init__.py]:
```suggestion
def get_compose_mapping(args=None):
    if args is None:
        args = sys.argv[1:]
    args = get_arguments(args=args)
```

File: src/dispatch/__init__.py [hunk 2]
```diff
@@ -380,29 +296,29 @@ def get_arguments():
...
         "--file",
         help=f"""{FormatText.bold}Mutually exclusive with respect to --latest and --cmd.{FormatText.end}
-        Specify a different location than the default {DEFAULT_TASKS_FILE} of the file containing request_id's, artifact URLs or request URLs.""",
+        Specify a different location than the default {dispatch_globals.DEFAULT_TASKS_FILE} of the file containing request_id's, artifact URLs or request URLs.""",
     )
     tasks_source.add_argument(
         "-c",
         "--cmd",
-        nargs="+",
+        action="append",
```
Comment (danmyway) [inline src/dispatch/__init__.py:310]: Could you please reference the change in the [README.md](https://github.com/danmyway/tesar/blob/devel/README.md) and [CHANGELOG.md](https://github.com/danmyway/tesar/blob/devel/CHANGELOG.md) files?

File: src/dispatch/__main__.py [hunk 3]
```diff
@@ -7,15 +7,16 @@
 
 from dispatch.__init__ import (
     ARTIFACT_MAPPING,
```
Comment (danmyway) [inline src/dispatch/__main__.py]:
```suggestion
from dispatch.dispatch_globals import (
    ARTIFACT_MAPPING,
```
Imports broken, please repoint to a correct path.
Same for PACKAGE_MAPPING and other files, where the globals are used.

**Review by danmyway — CHANGES_REQUESTED** [review]
Looks great, thank you!
Just minor changes to imports and README/CHANGELOG reference.

File: src/dispatch/__init__.py (hunk 2 above)
Comment (fernflower) [reply src/dispatch/__init__.py:310]:
```
You can specify a different path to the file with -f/--file or pass the jobs to get report for straight to the commandline with -c/--cmd.
The tool is able to parse and report for multiple variants of values as long as they are separated by a new-line (in the files) or a whitespace (on the commandline).
```
(from the [README](https://github.com/danmyway/tesar/blob/devel/README.md))
Should I do the same change for `-f` option?

Comment (fernflower) [reply src/dispatch/__init__.py:310]: Done

File: src/dispatch/__main__.py (hunk 3 above)
Comment (fernflower) [reply src/dispatch/__main__.py]: Done, sorry about that.

**Comment by fernflower** [comment]
So I have rebased against devel, verified the change manually and with unit tests.
```
(.venv) [ivasilev@localhost tesar]$ tesar report -c http://artifacts.osci.redhat.com/testing-farm/909143cf-89ca-4d62-8f81-959bf8ab4d03/ -c  http://artifacts.osci.redhat.com/testing-farm/14612a82-002d-4a8d-a5c4-613a0a75efeb/ 
Reporting for the requested tasks:
RHEL-8.6.0-Nightly ^(?!.*upgrade_plugin)(?!.*tier[2-3].*)(?!.*rhsm)(?!.*c2r)(?!.*sap)(?!.*7to8)(?!.*max_sst) https://api.dev.testing-farm.io/v0.1/requests/909143cf-89ca-4d62-8f81-959bf8ab4d03 ERROR
Request ended up in ERROR state, because Pipeline timeout expired., thus won't be reported.
See more details on the result page http://artifacts.osci.redhat.com/testing-farm/909143cf-89ca-4d62-8f81-959bf8ab4d03
Skipping to next request.
RHEL-8.8.0-Nightly tier1/upgrade_happy_path https://api.dev.testing-farm.io/v0.1/requests/14612a82-002d-4a8d-a5c4-613a0a75efeb COMPLETE
+--------------------------------------+--------------------+-------------------------------------+--------+
| UUID                                 | Target             | Test Plan                           | Result |
+--------------------------------------+--------------------+-------------------------------------+--------+
| 14612a82-002d-4a8d-a5c4-613a0a75efeb | RHEL-8.8.0-Nightly |                                     |        |
|                                      |                    | plans/8to9/tier1/upgrade_happy_path | PASSED |
+--------------------------------------+--------------------+-------------------------------------+--------+
```
```
[ivasilev@localhost tesar]$ tox
GLOB sdist-make: /home/ivasilev/projects/tesar/setup.py
py39 inst-nodeps: /home/ivasilev/projects/tesar/.tox/.tmp/package/1/tesar-2023.6.29.zip
py39 installed: exceptiongroup==1.1.2,iniconfig==2.0.0,packaging==23.1,pluggy==1.2.0,pytest==7.4.0,tesar @ file:///home/ivasilev/projects/tesar/.tox/.tmp/package/1/tesar-2023.6.29.zip,tomli==2.0.1
py39 run-test-pre: PYTHONHASHSEED='2760370913'
py39 run-test: commands[0] | pytest -s tests
============================================================================ test session starts ============================================================================
platform linux -- Python 3.9.15, pytest-7.4.0, pluggy-1.2.0
cachedir: .tox/py39/.pytest_cache
rootdir: /home/ivasilev/projects/tesar
collected 1 item                                                                                                                                                            

tests/test_args.py usage: pytest [-h] {test,report} ...
pytest: error: unrecognized arguments: http://artifacts.osci.redhat.com/testing-farm/14612a82-002d-4a8d-a5c4-613a0a75efeb/
.

============================================================================= 1 passed in 0.01s =============================================================================
__________________________________________________________________________________ summary __________________________________________________________________________________
  py39: commands succeeded
  congratulations :)
```

**Comment by danmyway** [comment]
@fernflower `src/dispatch/tf_send_request.py` still has import issues

**Comment by fernflower** [comment]
Latest github action run (with libkrb5 installed) - https://github.com/fernflower/tesar/actions/runs/5738124339/job/15551079675?pr=1

**Review by danmyway — APPROVED** [review]
Looks good, thank you!

**Pull request merged**